l10n_it_fatturapa_import_zip/models/attachment.py

```diff
@@ -0,0 +1,550 @@
+        return product
+
+    def adjust_accounting_data(self, product, line_vals):
+        wizard_import = self.env["wizard.import.fatturapa"]
+        if product.product_tmpl_id.property_account_income_id:
+            line_vals[
+                "account_id"
+            ] = product.product_tmpl_id.property_account_income_id.id
+        elif product.product_tmpl_id.categ_id.property_account_income_categ_id:
+            line_vals[
+                "account_id"
+            ] = product.product_tmpl_id.categ_id.property_account_income_categ_id.id
+        account = self.env["account.account"].browse(line_vals["account_id"])
+        new_tax = None
+        if len(product.product_tmpl_id.taxes_id) == 1:
+            new_tax = product.product_tmpl_id.taxes_id[0]
+        elif len(account.tax_ids) == 1:
+            new_tax = account.tax_ids[0]
+        if new_tax:
+            line_tax_id = line_vals.get("tax_ids") and line_vals["tax_ids"][0][2][0]
+            line_tax = self.env["account.tax"].browse(line_tax_id)
+            if new_tax.id != line_tax_id:
+                if line_tax and new_tax._get_tax_amount() != line_tax._get_tax_amount():
+                    wizard_import.log_inconsistency(
+                        _(
+                            "XML contains tax %s. Product %s has tax %s. Using "
+                            "the XML one"
+                        )
+                        % (line_tax.name, product.name, new_tax.name)
+                    )
+                else:
+                    # If product has the same amount of the one in XML,
+                    # I use it. Typical case: 22% det 50%
+                    line_vals["tax_ids"] = [(6, 0, [new_tax.id])]
+
+    def _set_invoice_lines(
+        self, product, invoice_line_data, invoice_lines, invoice_line_model
+    ):
+        if product:
+            invoice_line_data["product_id"] = product.id
+            self.adjust_accounting_data(product, invoice_line_data)
+        invoice_line_id = (
+            invoice_line_model.with_context(check_move_validity=False)
+            .create(invoice_line_data)
+            .id
+        )
+        invoice_lines.append(invoice_line_id)
+
+    def set_invoice_line_ids(self, FatturaBody, debit_account_id, partner, invoice):
+        invoice_lines = []
+        invoice_line_model = self.env["account.move.line"]
+        for line in FatturaBody.DatiBeniServizi.DettaglioLinee:
+            invoice_line_data = self._prepareInvoiceLine(debit_account_id, line)
+            invoice_line_data["move_id"] = invoice.id
+            product = self.get_line_product(line, partner)
+            self._set_invoice_lines(
+                product, invoice_line_data, invoice_lines, invoice_line_model
+            )
+        invoice.with_context(check_move_validity=False).update(
+            {"invoice_line_ids": [(6, 0, invoice_lines)]}
+        )
+
+    def invoiceCreate(self, fatt, fatturapa_attachment, FatturaBody, partner_id):
```

Comment:
anche qui viene introdotto molto codice duplicato in riferimento allo stesso metodo presente in [l10n_it_fatturapa_in](https://github.com/OCA/l10n-italy/blob/3ab6add597dc4d7357a13ff58181a783831abd2e/l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py#L966) Non c'è modo di ottimizzarlo?

Reply:
In generale vale quanto scrivo poco sopra: alcune funzioni sono riscritte poiché mentre su fatturapa_in è settato per fatture tipo `passive`, a me serve di importare fatture tipo `attive`